File: src/test/java/de/fraunhofer/aisec/crymlin/StructuresTest.java

package de.fraunhofer.aisec.crymlin;

import de.fraunhofer.aisec.analysis.structures.ConstantValue;
import de.fraunhofer.aisec.analysis.structures.ErrorValue;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class StructuresTest {

	@Test
	void testConstantValueEquals() {
		ConstantValue nullCV = ErrorValue.newErrorValue("narf");
		ConstantValue otherNullCV = ErrorValue.newErrorValue("other");
		ConstantValue oneCV = ConstantValue.of(1);
		ConstantValue otheroneCV = ConstantValue.of(1);
		ConstantValue twoCV = ConstantValue.of(2);

		assertNotEquals(nullCV, oneCV);
		assertNotEquals(nullCV, otherNullCV);

		assertNotEquals(oneCV, twoCV);
		assertEquals(oneCV, otheroneCV);

		assertNotEquals(oneCV, new Object());
	}

}

File: src/test/java/de/fraunhofer/aisec/codyze/legacy/crymlin/StructuresTest.kt
package de.fraunhofer.aisec.codyze.legacy.crymlin

import de.fraunhofer.aisec.codyze.legacy.analysis.*
import de.fraunhofer.aisec.codyze.legacy.analysis.resolution.ConstantValue
import kotlin.test.assertEquals
import kotlin.test.assertNotEquals
import org.junit.jupiter.api.*

internal class StructuresTest {
    @Test
    fun testConstantValueEquals() {
        val nullCV: ConstantValue = ErrorValue.newErrorValue("narf")
        val otherNullCV: ConstantValue = ErrorValue.newErrorValue("other")
        val oneCV = ConstantValue.of(1)
        val otheroneCV = ConstantValue.of(1)
        val twoCV = ConstantValue.of(2)
        assertNotEquals(nullCV, oneCV)
        assertNotEquals(nullCV, otherNullCV)
        assertNotEquals(oneCV, twoCV)
        assertEquals(oneCV, otheroneCV)
        assertNotEquals(oneCV, Any())
    }
}
